app/models/event.rb

```diff
@@ -78,33 +77,37 @@ def invalidate_schedule
   end
 
   def add_participant(user)
-    add_single_player_team(user)
-    participants << user
-
-    invalidate_schedule
+    team = user.create_single_team
+    add_team(team)
   end
 
-  def remove_participant(user)
-    remove_single_player_team(user)
-    participants.delete(user)
+  def has_participant?(user)
+    teams.each do |team|
+      if team.members.include?(user)
+        return true
+      end
+    end
+    false
+  end
 
-    invalidate_schedule
+  def ownes_participating_teams?(user)
+    (user.owned_teams & teams).present?
   end
 
-  def has_participant?(user)
-    participants.include?(user)
+  def available_team_slot?
```

Comment:
"team_slot_available" would indicate more strongly that this predicate method doesn't expect any arguments.